elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/engine/AbstractEntityHydrator.java

```diff
@@ -0,0 +1,198 @@
+        this.entityDictionary = entityDictionary;
+    }
+
+    /**
+     * Loads a map of relationship object ID to relationship object instance.
+     * <p>
+     * Note the relationship cannot be toMany. This method will be invoked for every relationship field of the
+     * requested entity. Its implementation should return the result of the following query
+     * <p>
+     * <b>Given a relationship {@code joinField} in an entity of type {@code entityClass}, loads all relationship
+     * objects whose foreign keys are one of the specified list, {@code joinFieldIds}</b>.
+     * <p>
+     * For example, when the relationship is loaded from SQL and we have the following example identity:
+     * <pre>
+     * {@code
+     * public class PlayerStats {
+     *     private String id;
+     *     private Country country;
+     *
+     *     @OneToOne
+     *     @JoinColumn(name = "country_id")
+     *     public Country getCountry() {
+     *         return country;
+     * }
+     * </pre>
+     * In this case {@code entityClass = PlayerStats.class}; {@code joinField = "country"}. If {@code country} is
+     * requested in {@code PlayerStats} query and 3 stats, for example, are found in database whose country ID's are
+     * {@code joinFieldIds = [840, 344, 840]}, then this method should effectively run the following query (JPQL as
+     * example)
+     * <pre>
+     * {@code
+     *     SELECT e FROM country_table e WHERE country_id IN (840, 344);
+     * }
+     * </pre>
+     * and returns the map of [840: Country(id:840), 344: Country(id:344)]
+     *
+     * @param entityClass  The type of relationship
+     * @param joinField  The relationship field name
+     * @param joinFieldIds  The specified list of join ID's against the relationshiop
+     *
+     * @return a list of hydrating values
+     */
+    protected abstract Map<Object, Object> getRelationshipValues(
+            Class<?> entityClass,
+            String joinField,
+            List<Object> joinFieldIds
+    );
+
+    public Iterable<Object> hydrate() {
+        //Coerce the results into entity objects.
+        MutableInt counter = new MutableInt(0);
+
+        List<Object> queryResults = getResults().stream()
+                .map((result) -> { return result instanceof Object[] ? (Object []) result : new Object[] { result }; })
+                .map((result) -> coerceObjectToEntity(result, counter))
+                .collect(Collectors.toList());
+
+        if (getStitchList().shouldStitch()) {
+            // relationship is requested, stitch relationship then
+            populateObjectLookupTable();
+            getStitchList().stitch();
+        }
+
+        return queryResults;
+    }
+
+    /**
+     * Coerces results from a {@link Query} into an Object.
+     *
+     * @param counter Monotonically increasing number to generate IDs.
+     * @return A hydrated entity object.
+     */
+    protected Object coerceObjectToEntity(Object[] result, MutableInt counter) {
```

Comment:
For the FiliDataStore I am working on, I would like this method to be abstract as the result fields returned from Fili might not be in the same order as the original request. The projects-result matching might not work.